shelley/chain-and-ledger/formal-spec/epoch.tex

```diff
@@ -810,6 +825,31 @@ \subsection{Complete Epoch Boundary Transition}
 
 The epoch transition rule calls $\mathsf{SNAP}$, $\mathsf{POOLREAP}$ and $\mathsf{NEWPP}$
 in sequence.
+Figure~\ref{fig:funcs:epoch-helper} gives some helper functions for the
+$\mathsf{EPOCH}$ transition.
+
+%%
+%% Figure - Helper Functions for Epoch Rules
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \fun{pvotes} \in \Update \to \Ppm \to (\VKeyGen \mapsto \Value) \\
+      & \fun{pvotes}~\var{up}~\var{ppm}
+      = \{k\mapsto v~\mid~ k\mapsto\var{ppm} ~\land~ \var{ppm}\mapsto v\}\\
+  \end{align*}
+
+  \begin{align*}
+      & \fun{tally} \in \Update \to \PParams\\
+      & \fun{tally}~\var{up} =
+          \Big\{\var{ppm}\mapsto v
```

Comment:
It seems `v` is not bound to anything on the rhs of the "pipe":

![image](https://user-images.githubusercontent.com/175315/59177336-11005f80-8b5c-11e9-873a-8af9654d9e87.png)

Reply:
it _could_ just be any `v` that type checks, but in practice it would be the values of some protocol parameter in the update. I'll try to reword it better...

Reply:
Hmmmm, but then we'd need an existential in the range of the set comprehension ... I guess `v` can only come from `range (range up)`, right? But anyway, I'll wait for the rewording :D